### PR TITLE
[IntroC] Proofread entire introduction to C section

### PR DESCRIPTION
**Pull request by PradyumnaShome**
**Is this a quick PR -- Typo, Quick Wording Fix, etc? If so, you don't need to follow the template -- please delete all and make the title of the PR representative of the changes and add a member of staff to review the PR. Thank you so much!**

## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I updated AUTHORS with my name and email.
- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] I updated/added relevant documentation.
- [x] The title of the PR is descriptive and doesn't have [WIP]
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

I don't understand what is meant by "plugin users".

### CODE REVIEW

File: introc/introc.tex [hunk 1]
```diff
@@ -781,45 +788,45 @@ \subsection{Everything is a File}
...
+Imagine that each of the file descriptors in the example actually refer to an entry in a table of objects that the operating system picks and chooses from (that is, the file descriptor table).
 Objects can be allocated and deallocated, closed and opened, etc.
-The program interacts with these objects by using the API or system calls specified
+The program interacts with these objects by using the API specified through system calls, and library functions.
 
 \subsection{System Calls}
 
 Before we dive into common C functions, we need to know what a system call is.
-If you are a student and completed HW0, feel free to gloss over this section.
+If you are a student and have completed HW0, feel free to gloss over this section.
```
Comment (bhuvy2) [inline introc/introc.tex]: Maybe just get rid of this sentence.

File: introc/introc.tex [hunk 2]
```diff
@@ -781,45 +788,45 @@ \subsection{Everything is a File}
...
 
-A system call is an operation that the kernel carries out instead of the program.
-The operating system prepares a system call then the kernel executes the system call to the best of its ability in kernel space.
-In the previous example we opened up a file descriptor object.
-We can now also write some bytes to the file descriptor object that represents a file and the operating system will do its best to get the bytes written to the disk.
+A system call is an operation that the kernel carries out.
+First, the operating system prepares a system call. Next, the kernel executes the system call to the best of its ability in kernel space, and is a privileged operation.
```
Comment (bhuvy2) [inline introc/introc.tex]: No comma for the `, and is a ..` that's a comma splice.

File: introc/introc.tex [hunk 3]
```diff
@@ -1055,7 +1062,7 @@ \subsection{string.h}
 	      Be careful though! Error handling is tricky because the function won't return an error code.
         If you give it a string that is not a number it will return 0.
         This means you can't differentiate between a valid ``0'' and an invalid string.
-        See the man page for more details on strol behavior with invalid and out of bounds values.
+        See the man page for more details on strtol behavior with invalid and out of bounds values.
```
Comment (bhuvy2) [inline introc/introc.tex]: `\keyword{strtol}`

File: introc/introc.tex [hunk 4]
```diff
@@ -1280,10 +1285,10 @@ \subsection{Places for strings}
 The only way to find out the length of a C string is to keep reading memory until you find the NULL byte.
 C characters are always exactly one byte each.
 
-\subsubsection{String constants are constant}
+\subsubsection{String literals are constant}
 
-A string constant is naturally constant.
-Any write will cause the operating system to produce a SEGFAULT.
+A string literal is naturally constant.
+Any write will cause the operating system to produce a SEGFAULT (and send the signal SIGSEGV or segmentation violation).
```
Comment (bhuvy2) [inline introc/introc.tex]: The parenthetical is redundant.

File: introc/introc.tex [hunk 5]
```diff
@@ -1355,35 +1361,35 @@ \subsubsection{String constants are constant}
 strcpy(ptr, "World"); // OK because now ptr is pointing to mutable memory (the array)
 \end{lstlisting}
 
-What to take away from this is that pointers * can point to any type of memory while the char arrays mentioned earlier will always refer to the same piece of memory.
+What one must take away from this, is that, pointers can point to any type of memory while the char arrays mentioned earlier will always refer to the same piece of memory.
```
Comment (bhuvy2) [inline introc/introc.tex]: We should probably just rewrite this sentence.

File: introc/introc.tex [hunk 6]
```diff
@@ -1355,35 +1361,35 @@ \subsubsection{String constants are constant}
...
 These addresses have numeric value, but usually we care about the contents underneath.
-In this section, we will try to take your through a very basic introduction of pointers.
+In this section, we will try to take you through a very basic introduction to pointers.
 
 \subsection{Pointer Basics}
 
 \subsubsection{Declaring a Pointer}
 
-A pointer refers to a memory address. The type of the pointer is useful -- it tells the compiler how many bytes need to be read/written and what the semantics for addition are.
+A pointer refers to a memory address. The type of the pointer is useful- it tells the compiler how many bytes need to be read/written and delineates the semantics for pointer arithmetic (addition and subtraction).
```
Comment (bhuvy2) [inline introc/introc.tex]: - won't get translated to a dash

File: introc/introc.tex [hunk 7]
```diff
@@ -1502,9 +1508,9 @@ \subsection{So what is a void pointer?}
 
 This does not require a cast because C automatically promotes \keyword{void*} to its appropriate type. \textbf{Note:}
 
-\keyword{gcc} and \keyword{clang} are not total ISO-C compliant, meaning that they will let you do arithmetic on a void pointer.
-They will treat it as a \keyword{char *} pointer.
-Do not do this because it may not work with all compilers!
+\keyword{gcc} and \keyword{clang} are not totally ISO C compliant, meaning that they will let you do arithmetic on a void pointer.
+They will treat it as a \keyword{char} pointer.
+Do not do this because it is not portable (it is not guaranteed to work with all compilers)!
```
Comment (bhuvy2) [inline introc/introc.tex]: portable -- it is not ...

File: introc/introc.tex [hunk 8]
```diff
@@ -1699,8 +1705,8 @@ \subsection{Undeclared or incorrectly prototyped functions}
 The system function `time' actually takes a parameter a pointer to some memory that can receive the time\_t structure or NULL.
 The compiler did not catch this error because the programmer did not provide a valid function prototype by including \keyword{time.h}.
 
-More confusingly this could compile, work for decades and then crash. The reason for that is that time would be found at link time, not compile time in the c standard library which almost surely is already in memory.
-Since we aren't passing a parameter in, we are hoping the arguments on the stack (any garbage) is zeroed out because if it isn't, time will try to write the result of the function to that garbage which will cause the program to SEGFAULT.
+More confusingly this could compile, work for decades and then crash. The reason for that is that time would be found at link time, not compile time in the C standard library which almost surely is already in memory.
+Since we aren't passing a parameter in, we are hoping the arguments on the stack (any garbage) is zeroed out because if it isn't, time will try to write the result of the function to that garbage which will cause the program to segfault.
```
Comment (bhuvy2) [inline introc/introc.tex]: we should probably have some consistency with SEGFAULT lower or upper case.

File: introduction/introduction.tex [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
 
-At the University of Illinois Urbana Champaign, I fundamentally believe that we have a right to make the university better for all students who are coming to the future.
+At the University of Illinois at Urbana-Champaign, I fundamentally believe that we have a right to make the university better for all students who enter in the future.
 It is a message etched into our Alma Mater and makes up the DNA of our course staffs.
 As such, we created the coursebook.
 The coursebook is a free and open systems programming textbook that anyone can read, contribute to, and modify for now and forever.
-We don't think information should be behind a walled garden, and we truly believe that complex concepts can be explained simply and fully for anyone to understand.
-The goal of this book is to teach you the basics and some basic intuition about the complexities of systems programming.
+We don't think information should be behind a walled garden, and we truly believe that complex concepts can be explained simply and fully, for anyone to understand.
+The goal of this book is to teach you the basics f and give you some intuition into the complexities of systems programming.
```
[the remark this answers is not in the document]
Comment (PradyumnaShome) [reply introduction/introduction.tex]: Typo, now fixed.

File: introc/introc.tex [hunk 10]
```diff
@@ -7,46 +7,46 @@ \chapter{C Programming Language}
...
-    C let's you manage your memory.
-    This can also bite you if you have memory errors.
-	\item It's Everywhere.
-    Pretty much every computer that is not embedded has some way of interfacing with C.
+	\item Speed. There is very little separating you and the system.
+	\item Simplicity.
+    C and its standard library comprise a simple set of portable functions.
+	\item Manual Memory Management.
+    C gives you the ability you manage your memory.
+    However, this can bite you, if you have memory errors.
```
[the remark this answers is not in the document]
Comment (PradyumnaShome) [reply introc/introc.tex]: All right!

File: introc/introc.tex [hunk 11]
```diff
@@ -7,46 +7,46 @@ \chapter{C Programming Language}
...
+	\item Ubiquity.
+    Almost every computer that is not embedded has some way to interface with C.
```
[the remark this answers is not in the document]
Comment (PradyumnaShome) [reply introc/introc.tex]: I have made it more specific, and mentioned FFI.

File: introc/introc.tex (hunk 5 above)
Comment (PradyumnaShome) [reply introc/introc.tex]: Done.

File: introc/introc.tex (hunk 6 above)
Comment (PradyumnaShome) [reply introc/introc.tex]: Oh, I see. Reverted.

**Comment by bhuvy2** [comment]
Is this good to merge? #83 depends on this I presume

**Pull request merged**